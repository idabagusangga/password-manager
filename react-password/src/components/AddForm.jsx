import React, { Component } from 'react';
import { bindActionCreators } from 'redux'
import {  connect } from 'react-redux'
import { ADD_PASSWORD } from '../store/actions/passwords'
import Spinner from './LoadingSpin'
import { Form, 
        Input, 
        Tooltip, 
        Icon, 
        Cascader, 
        Select, 
        Row, 
        Col, 
        Checkbox, 
        Button, 
        AutoComplete } from 'antd';
        
const FormItem = Form.Item;
const Option = Select.Option;
const AutoCompleteOption = AutoComplete.Option;
const residences = []
const lowerCase = new RegExp ("^(?=.*[a-z])")
const upperCase = new RegExp ("^(?=.*[A-Z])")
const numeric = new RegExp ("^(?=.*[0-9])")
const length = new RegExp ("^(?=.{8,})")


class AddForm extends Component {
  state = {
    confirmDirty: false,
    autoCompleteResult: [],
  };
  handleSubmit = (e) => {
    e.preventDefault();
    this.props.form.validateFieldsAndScroll((err, values) => {
      if (!err) {
        this.props.ADD_PASSWORD(values)
      }
    });
  }
  handleConfirmBlur = (e) => {
    const value = e.target.value;
    this.setState({ confirmDirty: this.state.confirmDirty || !!value });
  }
  compareToFirstPassword = (rule, value, callback) => {
    const form = this.props.form;
    if (value && value !== form.getFieldValue('password')) {
      callback('Two passwords that you enter is inconsistent!');
    } else {
      callback();
    }
  }
  validateLowerCase = (rule, value, callback) => {
    const form = this.props.form;
    if (value && !lowerCase.test(value) ) {
      callback('Must contain at least 1 lowercase alphabetical character')
    }
  }
  validateToNextPassword = (rule, value, callback) => {
    const form = this.props.form;
    if (value && !lowerCase.test(value) ) {
      callback('Must contain at least 1 lower case alphabetical character')
    } else if (value && !upperCase.test(value) ) {
      callback('Must contain at least 1 upper case alphabetical character')
    } else if (value && !numeric.test(value) ) {
      callback('Must contain at least 1 numeric case')
    } else if (value && !length.test(value) ) {
      callback('Password must be 8 characters or longer')
    }
    else if (value && this.state.confirmDirty) {
      form.validateFields(['confirm'], { force: true });
    }
    callback();
  }
  handleWebsiteChange = (value) => {
    let autoCompleteResult;
    if (!value) {
      autoCompleteResult = [];
    } else {
      autoCompleteResult = ['.com', '.org', '.net'].map(domain => `${value}${domain}`);
    }
    this.setState({ autoCompleteResult });
  }
  render() {
    const { getFieldDecorator } = this.props.form;
    const { autoCompleteResult } = this.state;

    const formItemLayout = {
      labelCol: {
        xs: { span: 24 },
        sm: { span: 8 },
      },
      wrapperCol: {
        xs: { span: 24 },
        sm: { span: 16 },
      },
    };
    const tailFormItemLayout = {
      wrapperCol: {
        xs: {
          span: 24,
          offset: 0,
        },
        sm: {
          span: 16,
          offset: 8,
        },
      },
    };
    const prefixSelector = getFieldDecorator('prefix', {
      initialValue: '86',
    })(
      <Select style={{ width: 70 }}>
        <Option value="86">+86</Option>
        <Option value="87">+87</Option>
      </Select>
    );

    const websiteOptions = autoCompleteResult.map(website => (
      <AutoCompleteOption key={website}>{website}</AutoCompleteOption>
    ));
    if (this.props.dataStatus.isLoading === true) {
      return (
        <Spinner></Spinner>
      )
    } else {
      return (
        <Form onSubmit={this.handleSubmit}>
          <FormItem
            {...formItemLayout}
            label="E-mail"
          >
            {getFieldDecorator('email', {
              rules: [{
                type: 'email', message: 'The input is not valid E-mail!',
              }, {
                required: true, message: 'Please input your E-mail!',
              }],
            })(
              <Input />
            )}
          </FormItem>
          <FormItem
            {...formItemLayout}
            label="Password"
          >
            {getFieldDecorator('password', {
              rules: [{
                required: true, message: 'Please input your password!',
              }, {
                validator: this.validateToNextPassword,
              }],
            })(
              <Input name="password" type="password" />
            )}
          </FormItem>
          <FormItem
            {...formItemLayout}
            label="Confirm Password"
          >
            {getFieldDecorator('confirm', {
              rules: [{
                required: true, message: 'Please confirm your password!',
              }, {
                validator: this.compareToFirstPassword,
              }],
            })(
              <Input type="password" onBlur={this.handleConfirmBlur} />
            )}
          </FormItem>
          <FormItem
            {...formItemLayout}
            label="Website"
          >
            {getFieldDecorator('website', {
              rules: [{ required: true, message: 'Please input website!' }],
            })(
              <AutoComplete
                dataSource={websiteOptions}
                onChange={this.handleWebsiteChange}
                placeholder="website"
              >
                <Input />
              </AutoComplete>
            )}
          </FormItem>
          <FormItem
            {...formItemLayout}
            label="PIN"
            extra="This is used to retrieve, edit and delete your password later"
          >
            <Row gutter={8}>
              <Col span={12}>
                {getFieldDecorator('PIN', {
                  rules: [{ required: true, message: 'Please enter your PIN' }],
                })(
                  <Input type="password" />
                )}
              </Col>
            </Row>
          </FormItem>
           <Button type="primary" htmlType="submit">{this.props.dataStatus.button}</Button>
        </Form>
      );  
    }
  }
}

const mapStateToProps = (state) => {
  return {
    dataStatus: state.passwordReducer
  }
}

const mapDispatchToProps = dispatch => bindActionCreators ({
  ADD_PASSWORD
},dispatch)


const WrappedRegistrationForm = Form.create()(AddForm);

export default connect(mapStateToProps,mapDispatchToProps)(WrappedRegistrationForm);