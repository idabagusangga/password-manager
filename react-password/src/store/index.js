import { createStore, combineReducers, applyMiddleware, compose } from 'redux'
import { userReducer } from './reducers/userReducer'
import { passwordReducer } from './reducers/passwordReducer'

import firebase from 'firebase'
import 'firebase/firestore'
import thunk from 'redux-thunk'


const rootReducers = combineReducers({
  userReducer,
  passwordReducer
})

const store = createStore(rootReducers,
   /* preloadedState, */
   window.__REDUX_DEVTOOLS_EXTENSION__ && window.__REDUX_DEVTOOLS_EXTENSION__(),
   applyMiddleware(thunk))

export default store