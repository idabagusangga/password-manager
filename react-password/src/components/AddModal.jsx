import React, { Component } from 'react';
import { Modal, Button } from 'antd';
import AddForm from './AddForm'
import {  connect } from 'react-redux'
import { bindActionCreators } from 'redux'


//modalState must be in redux


class AddModal extends Component {
    state = {
     ModalText: 'Content of the modal',
     visible: false,
     confirmLoading: false,
   }
   showModal = () => {
     this.setState({
       visible: true,
     });
   }
   handleOk = () => {
     this.setState({
       ModalText: 'The modal will be closed after two seconds',
       confirmLoading: true,
     });
     setTimeout(() => {
       this.setState({
         visible: false,
         confirmLoading: false,
       });
     }, 2000);
     // this.props.RESET_ADD_PASSWORD()
   }
   handleCancel = () => {
     console.log('Clicked cancel button');
     this.setState({
       visible: false,
     });
   }
   render() {
     const { visible, confirmLoading, ModalText } = this.state;
     return (
       <div>
         <Button type="primary" onClick={this.showModal}>Open</Button>
         <Modal title="Title"
           visible={visible}
           onOk={this.handleOk}
           confirmLoading={confirmLoading}
           onCancel={this.handleCancel}
         >
           <AddForm></AddForm>
         </Modal>
       </div>
     );
   }
  }

const mapDispatchToProps = dispatch => bindActionCreators ({
},dispatch)

export default connect(null,mapDispatchToProps)(AddModal);