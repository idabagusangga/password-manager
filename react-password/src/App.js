import React, { Component } from 'react';
import { BrowserRouter, Route, Link } from 'react-router-dom'
import Home from './components/home'
import Edit from './components/Edit'
import logo from './logo.svg';
import './App.css';
import store from './store/index'
import { Provider } from 'react-redux'
import { Layout, Menu, Icon } from 'antd'
const { Sider, Footer, Content } = Layout;


class App extends Component {
  constructor () {
    super ()
    this.state = {
      collapsed: false
    }
  }
  toggle = () => {
    this.setState({
      collapsed: !this.state.collapsed
    })
  }
  render() {
    return (
      <Provider store = { store }>    
        <Layout>
          <Sider
            breakpoint="lg"
            collapsedWidth="0"
            onCollapse={(collapsed, type) => { console.log(collapsed, type); }}
          >
            <div className="logo" />
            <Menu theme="dark" mode="inline" defaultSelectedKeys={['1']}>
              <Menu.Item key="1">
                <Icon type="user" />
                <span className="nav-text">Home</span>
              </Menu.Item>
              <Menu.Item key="2">
                <Icon type="lock" />
                <span className="nav-text">Passwords</span>
              </Menu.Item>
              <Menu.Item key="3">
                <Icon type="key" />
                <span className="nav-text">Add Password</span>
              </Menu.Item>
              <Menu.Item key="4">
                <Icon type="user" />
                <span className="nav-text">Logout</span>
              </Menu.Item>
            </Menu>
          </Sider>
          <Layout>
            <div className="App">
            <Content style={{ margin: '24px 16px 0' }}>
              <div style={{ padding: 24, background: '#fff', minHeight: 360 }}>
                <img src={logo} className="App-logo" alt="logo" />
                <h1 className="App-title">Welcome to React Password Manager</h1>
                <BrowserRouter>
                <div>
                  <Route exact path="/" component={Home}></Route>
                  <Route path='/edit/:name' component={Edit}></Route>
                </div>

                </BrowserRouter>
              </div>
            </Content>
            </div>
            <Footer style={{ textAlign: 'center' }}>
              React Password Manager @idabagusangga
            </Footer>
          </Layout>
        </Layout>
     </Provider>
    );
  }
}

export default App;
