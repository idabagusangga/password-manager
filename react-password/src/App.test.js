import React from 'react';
import ReactDOM from 'react-dom';
import App from './App';
import Home from './components/home'
import Enzyme from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import { shallow, mount, render} from 'enzyme'
import Jumbotron from './components/carousel'
import AddModal from './components/AddModal'
import DataTable from './components/Tables'
import AddForm from './components/AddForm'
import store from './store/index'
import Spinner from './components/LoadingSpin'
import { Provider } from 'react-redux'
import { Carousel, Layout } from 'antd'
import { Form, 
        Input, 
        Tooltip, 
        Icon, 
        Cascader, 
        Select, 
        Row, 
        Col, 
        Checkbox, 
        Button, 
        AutoComplete } from 'antd';
import renderer from 'react-test-renderer'
const { Sider, Footer, Content } = Layout;
import { BrowserRouter, Route, Link } from 'react-router-dom'


//testing tambahain sama function testing

Enzyme.configure({
  adapter: new Adapter()
})


describe ('<Home/>', () => {
  it('should render div containing Jumbotron, table, and modal', () => {
    const wrapper = shallow ((
      <Home/>
    ))
    expect(wrapper.contains(<div><Jumbotron></Jumbotron><AddModal></AddModal><DataTable></DataTable></div>)).toBe(true)
  })
})

describe ('<Carousel/>', () => {
  it('should render Carousel', () => {
    const wrapper = shallow ((
      <Jumbotron></Jumbotron>
    ))
    console.log('here: ======> ', wrapper.find('div'))
    expect(wrapper.find('Carousel')).toHaveLength(1)
    expect(wrapper.find('div')).toHaveLength(4)
    expect(wrapper.find('h2')).toHaveLength(4)
  })
})

describe ('Store', () => {
  const wrapperStore = mount(
    <Provider store = { store }>
      <Home></Home>
    </Provider>
  )
  it('Should load initialState', () => {
    expect(wrapperStore.props().store.getState().passwordReducer).toEqual({
      data: [],
      isLoading: false,
      error: false,
      button: 'Register'
    })
  })
})

describe ('<AddForm/>', () => {
  const wrapperApp = shallow (<AddForm/>, {context: {store}}).dive().dive()
  it('should render Form', () => {
    expect(wrapperApp.find('Form')).toBeTruthy()
  })
  it('Should render Submit Button', () => {
    expect(wrapperApp.find('Button')).toBeTruthy()
  })
  it('Should render FormItem', () => {
    expect(wrapperApp.find('FormItem')).toBeTruthy()
  })
  it('Should warn if password does not contain Upper-case', () => {
    expect(wrapperApp.find('Input')).toHaveLength(5);
    wrapperApp.find('Input').at(1).simulate('change', {
      target: {
        name: 'password',
        value: 'aiue1233'
      }
    })
    console.log(wrapperApp.find('Input').at(3).text());
    expect(wrapperApp.find('.ant-form-explain').at(1).text()).toMatch('Must contain at least 1 lower case alphabetical character')
  })
})

describe('snapshot testing', () => {
  it('rendered app must be the same as snapshot', () => {
    let tree = renderer.create(<Spinner/>)
    expect(tree).toMatchSnapshot()
  })
})
