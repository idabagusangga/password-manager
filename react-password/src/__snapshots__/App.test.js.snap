// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot testing rendered app must be the same as snapshot 1`] = `
<div
  style={
    Object {
      "marginLeft": "200px",
    }
  }
>
  <div
    className="ant-spin ant-spin-lg ant-spin-spinning ant-spin-show-text"
  >
    <span
      className="ant-spin-dot ant-spin-dot-spin"
    >
      <i />
      <i />
      <i />
      <i />
    </span>
  </div>
  <br />
  Please Wait...
</div>
`;
